resotoclient/__init__.py

```diff
@@ -372,20 +403,167 @@ def cli_evaluate(
 
-    def cli_execute(
-        self, command: str, graph: str = "resoto", **env: str
-    ) -> Iterator[JsValue]:
-        props = {"graph": graph, "section": "reported", **env}
+    def cli_execute_raw(
+        self,
+        command: str,
+        graph: Optional[str] = "resoto",
+        section: Optional[str] = "reported",
+        headers: Optional[Dict[str, str]] = None,
+        files: Optional[Files] = None,
+        **env: str,
+    ) -> requests.Response:
+        props: Dict[str, str] = {}
+        if graph:
+            props["graph"] = graph
+        if section:
+            props["section"] = section
+
+        body: Optional[Any] = None
+        headers = headers or {}
+        if not files:
+            headers["Content-Type"] = "text/plain"
+            body = command
+        else:
+            headers["Resoto-Shell-Command"] = command
+            headers["Content-Type"] = "multipart/form-data; boundary=file-upload"
+            parts = {
+                name: (name, open(path, "rb"), "application/octet-stream")
+                for name, path in files.items()
+            }
+            body = MultipartEncoder(parts, "file-upload")
 
         response = self._post(
             f"/cli/execute",
-            data=command,
+            data=body,
             params=props,
-            headers={"Content-Type": "text/plain"},
+            headers=headers,
             stream=True,
         )
+        return response
+
+    def cli_execute(
+        self,
+        command: str,
+        graph: Optional[str] = "resoto",
+        section: Optional[str] = "reported",
+        headers: Optional[Dict[str, str]] = None,
+        files: Optional[Files] = None,
+        **env: str,
+    ) -> Iterator[JsValue]:
+        """
+        Execute a CLI command and return the result as a stream of text or JSON objects.
+
+        Binary or multi-part responses will trigger an exception.
+        """
+
+        response = self.cli_execute_raw(
+            command=command,
+            graph=graph,
+            section=section,
+            headers=headers,
+            files=files,
+            **env,
+        )
+
         if response.status_code == 200:
-            return map(lambda l: json_loadb(l), response.iter_lines())
+            content_type = response.headers.get("Content-Type")
+            if content_type == "text/plain":
+                return iter([response.text])
+            elif content_type == "application/json":
+                return iter([response.json()])
+            elif content_type == "application/x-ndjson":
+                return map(lambda line: json_loadb(line), response.iter_lines())
+            else:
+                raise NotImplementedError(
+                    f"Unsupported content type: {content_type}. Use cli_execute_raw or cli_execute_with_callback instead."
+                )
+        else:
+            raise AttributeError(response.text)
+
+    def cli_execute_bytes(
```

Comment:
which use case is covered here? If I want raw bytes, I could use `ececute_raw` or?